src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -346,14 +343,7 @@ private HttpClient getHttpClient(PrintStream logger, AbstractBuild<?, ?> build)
         String stashServer = stashServerBaseUrl;
         DescriptorImpl descriptor = getDescriptor();
 
-		// Determine if we are using the local or global settings
-		String credentialsId = getCredentialsId();
-		if (StringUtils.isBlank(credentialsId)) {
-			credentialsId = descriptor.getCredentialsId();
-		}
-
-		Credentials credentials = CredentialsMatchers.firstOrNull(CredentialsProvider.lookupCredentials(CertificateCredentials.class,
-				Jenkins.getInstance(), ACL.SYSTEM), CredentialsMatchers.withId(credentialsId));
+        CertificateCredentials certificateCredentials = getCredentials(CertificateCredentials.class, build.getProject());
```

Comment:
Yeah, I've noticed you haven't change much now. It's ok then. I'll consider review it later or reimplement when 2.0 comes